pymobiledevice3/services/mobile_config.py

```diff
@@ -23,6 +27,17 @@ def hello(self) -> None:
     def flush(self) -> None:
         self._send_recv({'RequestType': 'Flush'})
 
+    def escalate(self, pkcs12: bytes, password: str) -> None:
+        decrypted_p12 = load_pkcs12(pkcs12, password.encode('utf-8'))
+
+        escalate_response = self._send_recv({'RequestType': 'Escalate', 'SupervisorCertificate': decrypted_p12.cert.certificate.public_bytes(Encoding.DER)})
+        try:
+            signed_challenge = PKCS7SignatureBuilder().set_data(escalate_response['Challenge']).add_signer(decrypted_p12.cert.certificate, decrypted_p12.key, hashes.SHA256()).sign(Encoding.DER, [])
+            self._send_recv({'RequestType': 'EscalateResponse', 'SignedRequest': signed_challenge})
+            self._send_recv({'RequestType': 'ProceedWithKeybagMigration'})
+        except TypeError as e:
```

Comment:
Because you suggested it here: https://github.com/doronz88/pymobiledevice3/pull/472#discussion_r1233116565
But I think I misunderstood you, the type error was raised by `escalate_response.get('Chellange')` and now that I changed it to index-access, I should catch an `IndexError`, shouldn’t I?

Reply:
Since you changed it to access `Challenge` with `[]` it will not raise `TypeError` but `KeyError`.

More than that - It seems like you are pretty sure `Challenge` will always exist in `escalate_response` - if this is the case you can remove the try except completely

Reply:
done